src/pal/inc/unixasmmacrosarm.inc

```diff
@@ -177,14 +180,18 @@ C_FUNC(\Name\()_End):
 .macro EPILOG_WITH_TRANSITION_BLOCK_TAILCALL
 
         .if (__PWTB_SaveFPArgs == 1)
-                add r6, sp, #(__PWTB_FloatArgumentRegisters)
+                mov r6, sp
                 vldm r6, {d0-d7}
         .endif
 
-        free_stack __PWTB_StackAlloc
-        POP_CALLEE_SAVED_REGISTERS
-        POP_ARGUMENT_REGISTERS
-        
+	add r6, sp, #((__PWTB_StackAlloc) + 9 * 4)
+	ldm r6, {r0 - r3}
+
+	add r6, sp, #(__PWTB_StackAlloc)
+	ldm r6, {r4 - r11, lr}
```

Comment:
Unfortunately for me, the stack gets messed up right after this ldm instruction. After that, the callstack is always messed up for me.